src/pyfmi/fmi_algorithm_drivers.py

```diff
@@ -1091,7 +1107,9 @@ def solve(self):
             final_time = t+h
 
             start_time_point = timer()
-            result_handler.integration_point()
+            # down-sampling of result; step starts at 0
+            if ((step + 1) % self.result_downsampling_factor == 0) or ((step + 1) == self.ncp):
```

Comment:
I effectively inlined what was previously a function, since this would be the only placed used.